qingstor/sdk/build.py

```diff
@@ -181,3 +184,46 @@ def parse_request_uri(self):
                 (scheme, netloc, path, params, "", fragment)
             ) + "?" + "&".join(sorted(query))
         return parsed_uri
+
+    def parse_request_virtual_style_uri(self):
+        properties = self.parse_request_properties()
+        zone = properties.get("zone", "")
+        self.config.zone = zone
+        port = str(self.config.port)
+        request_uri = self.operation["URI"]
+        filed = request_uri.split("/", 3)
+        domain = self.config.host
+        if zone != "":
+            domain = "".join([
+                zone, ".", domain
+            ])
+        if len(filed) >= 2 and filed[1] != "":
+            domain = "".join([
+                filed[1], ".", domain
+            ])
+            self.operation["Headers"]["Host"] = domain
+        parsed_uri = "".join([
+            self.config.protocol, "://", domain, ":", port
+        ])
+        if len(filed) == 3 and filed[2] != "":
+            parsed_uri = "".join([
+                parsed_uri, "/", filed[2]
+            ])
+        if len(properties):
+            for (k, v) in properties.items():
+                parsed_uri = parsed_uri.replace("<%s>" % k, v)
+                self.operation["Headers"]["Host"] = self.operation["Headers"]["Host"].replace("<%s>" % k, v)
```

Comment:
We could imporve the readability via extract a `get_host` function and set Host just before sending requests (because `Host` will not be included in sign).